src/components/Quests/UnlockQuest.js

```diff
@@ -0,0 +1,10 @@
+export function unlockQuest(quests, questId, setQuests) {
+	const updatedQuests = quests.map((quest) => {
+		if (quest.id === questId + 1) {
+			return { ...quest, isUnlocked: true }
+		} else {
+			return quest
+		}
+	})
+	setQuests(updatedQuests)
+}
```

Comment:
This logic should not be here. It should be handled by a condition in the markQuestAsCompleted method of the QuestsProvider.